### ISSUE TITLE
Extension activation fails with 'Error: Failed to find up to date dotnet cli on the path.'

### ISSUE THREAD

**Issue by nickmccurdy** [OmniSharp/omnisharp-vscode#1899]
## Environment data
`dotnet --info` output:
```

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.3
  Build    : a9190d4a75f4a982ae4b4fa8d1a24526566c69df

```
VS Code version: 1.18.1
C# Extension version: 1.13.1

## Steps to reproduce
1. Install .NET Core **Runtime** (NOT SDK): [link](https://www.microsoft.com/net/download/thank-you/dotnet-runtime-2.0.3-macos-x64-installer)
2. Install VSCode and the extension
3. Relaunch VSCode

## Expected  behavior
VSCode should detect the dotnet CLI because it's automatically added to my path. Running `dotnet` in an ordinary terminal confirms that it's installed on my path.

## Actual behavior
Error: Failed to find up to date dotnet cli on the path.

**Comment by gregg-miskelly** [comment]
@nickmccurdy all the C# extension does is call `dotnet`. So this isn't a problem with C#, but maybe we can help troubleshoot.

First, if you open a VS Code integrated terminal, do you see the same issue?
There are some troubleshooting tips in https://github.com/Microsoft/vscode/issues/21655 that you may find helpful.

**Comment by nickmccurdy** [comment]
Thanks for the information. Actually, I was mistaken, I didn't install this via Homebrew, so I updated my issue. I'm not 100% sure but I think I installed it via the Unity installer, so it might be Mono if it provides a `dotnet` command. However, the command is actually found in VSCode's terminal.

**Comment by sourcesoft** [comment]
having the same issue:
Displays the error after trying to debug.

```
launch: program '/usr/local/share/dotnet' does not exist.
```
<img width="1199" alt="screen shot 2017-12-01 at 6 46 10 pm" src="https://user-images.githubusercontent.com/608906/33508392-0d3710be-d6c8-11e7-835e-5c8816590255.png">

I can run `dotnet` from internal terminal or OSX terminal.

**Comment by gregg-miskelly** [comment]
@sourcesoft please open a new issue. Your issue sounds different.

**Comment by gregg-miskelly** [comment]
@nickmccurdy to make sure I understand, can you tell me exactly what error you get?

**Comment by nickmccurdy** [comment]
```
Error: Failed to find up to date dotnet cli on the path.
```

**Comment by gregg-miskelly** [comment]
@nickmccurdy Thanks. I went off in the wrong direction. It doesn't sound like an error finding dotnet. Something else is going wrong.

First, to make sure I am looking at the right code. You are seeing this when you start the extension (ex: open a workspace with a .cs file). Correct?

**Comment by gregg-miskelly** [comment]
Also, is this really the full text you see from `dotnet --info`?

```

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.3
  Build    : a9190d4a75f4a982ae4b4fa8d1a24526566c69df

```

**Comment by gregg-miskelly** [comment]
Assuming that is your complete output from `dotnet --info` then I believe I understand what is going on.

You have the .NET Core run time installed, but no the SDK, and the code [here](https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/coreclr-debug/util.ts#L108) handles this wrong.

If you don't care about .NET Core debugging, you can just ignore the message until we get this fixed. Alternatively, if you install the .NET Core SDK ([instructions](https://www.microsoft.com/net/learn/get-started/macos)) this will go away.

**Comment by nickmccurdy** [comment]
Thanks for all the info! Are you suggesting that the error is a false positive and the extension should already be fully working for me, or do I need to install the SDK if I want the full functionality?

> You are seeing this when you start the extension (ex: open a workspace with a .cs file). Correct?
Yes, whenever I open a `.cs` file with the extension installed.

**Comment by gregg-miskelly** [comment]
Great!

If you are trying to do .NET Core development - you should install the .NET SDK. 
If you are doing Unity/Mono - I don't think there is any reason why you would need the .NET Core SDK. So as long as this warning isn't annoying you too much you can just ignore it.

**Comment by nickmccurdy** [comment]
Thanks, I'll just disable it unless I decide I need to do .NET Core development or debugging. 😄 

Let's keep this issue open if you want to track the false positive error.

**Comment by jwvanderbeck** [comment]
I see this issue is old, but it isn't closed and I am having pretty much the same issue.

I got an error saying the dotnet CLI weasn't installed along with a link to download it from.  I followed that link and installed it, and continue to get the error.

Opening the VS Code Terminal and running `dotnet --info` indeed returns an error.  If I open CMD and run `dotnet --info` however I get output:

``
C:\Users\jwvan>dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.401
 Commit:    91b1c13032

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.401\

Host (useful for support):
  Version: 2.1.3
  Commit:  124038c13e

.NET Core SDKs installed:
  2.1.401 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
``

**Comment by jwvanderbeck** [comment]
VSCode was restarted after installing the CLI tools yes.

**Comment by gregg-miskelly** [comment]
@jwvanderbeck your issue isn't related to the original issue, and also is probably a VS Code issue rather than an issue with this extension. But we can see if we can help.

How are you starting VS Code (ex: start menu, command prompt, etc)?
If you compare the output from `echo %PATH%` in the VS Code integrated terminal vs. a normal command prompt, what differences do you see?
If you look at your PATH in a normal command prompt, do you notice anything odd about it? (ex: special characters)
Are you using any sort of replacement command prompt?

**Comment by jwvanderbeck** [comment]
> How are you starting VS Code (ex: start menu, command prompt, etc)?

From the Start Menu icon that the installer created.  VS Code is installed with the newer user install versus the old system install if that matters

> If you compare the output from echo %PATH% in the VS Code integrated terminal vs. a normal command prompt, what differences do you see?

Output shown below, but it looks like the PATH variable is empty in VSCode.  In CMD it looks fine.  Actually I just checked in Power Shell directly and it is missing in there as well, so might be a Powershell issue.

> If you look at your PATH in a normal command prompt, do you notice anything odd about it? (ex: special characters)

No

> Are you using any sort of replacement command prompt?

No

Looks like as you stated this isn't an issue with the plugin, but either a VSCode or Powershell or other system issue so I will dig into that on my side.  Sorry for the noise.

VSCode Terminal:
```
PS C:\Users\jwvan> echo $PATH
PS C:\Users\jwvan> echo %PATH%
%PATH%
```
CMD:
```
C:\Users\jwvan>echo %PATH%
C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\libnvvp;C:\Program Files\Oculus\Support\oculus-runtime;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Users\jwvan\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Git LFS;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Microsoft VS Code\bin;C:\Program Files\dotnet\;C:\Users\jwvan\AppData\Local\Programs\Python\Python36-32\Scripts\;C:\Users\jwvan\AppData\Local\Programs\Python\Python36-32\;C:\Users\jwvan\AppData\Local\Microsoft\WindowsApps;C:\Users\jwvan\AppData\Local\GitHubDesktop\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\bin;C:\Users\jwvan\AppData\Roaming\npm;C:\Program Files\Heroku\bin;C:\Program Files\Microsoft VS Code\bin;C:\Users\jwvan\AppData\Local\Microsoft\WindowsApps;
```

**Comment by jwvanderbeck** [comment]
A little digging online and I found that in powershell the proper command is actually `echo $env:PATH` and that does work for me in both PS and VSCode:

```
PS C:\Users\jwvan> echo $env:PATH
C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\libnvvp;C:\Program Files\Oculus\Support\oculus-runtime;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Users\jwvan\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Git LFS;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Users\jwvan\AppData\Local\Programs\Python\Python36-32\Scripts\;C:\Users\jwvan\AppData\Local\Programs\Python\Python36-32\;C:\Users\jwvan\AppData\Local\Microsoft\WindowsApps;C:\Users\jwvan\AppData\Local\GitHubDesktop\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\bin;C:\Users\jwvan\AppData\Roaming\npm;C:\Program Files\Heroku\bin;C:\Program Files\Microsoft VS Code\bin;C:\Users\jwvan\AppData\Local\Microsoft\WindowsApps;;C:\Users\jwvan\AppData\Local\Programs\Microsoft VS Code\bin
```

**Comment by gregg-miskelly** [comment]
@jwvanderbeck To make sure I understand, are you saying that you get exactly the same PATH in both the VS Code integrated terminal, and a regular PS prompt, but that in a regular PS prompt the `dotnet --info` command works, but it does NOT work in the integrated terminal?

**Comment by jwvanderbeck** [comment]
Ok I have it working now.  Not sure why it was giving me trouble originally but uninstalling the plugin, restarting VSCode, then reinstalling the plugin seems to have fixed the problem.

Maybe some quirk of me installing the plugin originally before I have installed the dotnet cli tools.

**Comment by rchande** [comment]
@jwvanderbeck Glad you got it working, but that is a bit strange--the C# extension shouldn't maintain state (aside from options) across reloads.

**Comment by renns** [comment]
I was having this issue as well. I figured I'd share in case anyone else comes here to find an answer.

This page helped me: https://github.com/OmniSharp/omnisharp-vscode/wiki/Troubleshooting:-'The-.NET-Core-SDK-cannot-be-located.'-errors

My particular problem was that I had an old 32-bit version of the dotnet cli installed, and that was before the 64-bit on my PATH. I removed the 32-bit version from the PATH environment variable and then everything worked.
